src/core/Akka/Actor/ActorSelection.cs

```diff
@@ -213,26 +226,35 @@ public override bool Equals(object obj)
 
         protected bool Equals(ActorSelection other)
         {
-            return Equals(Anchor, other.Anchor) && Equals(Path, other.Path);
+            return Equals(Anchor, other.Anchor) && Equals(PathString, other.PathString);
```

Comment:
`Path` was checked for reference equality, so two selections were only equal if they used the same path array reference... Used `PathString` for simplicity instead